src/hotspot/os/aix/os_aix.cpp

```diff
@@ -461,7 +461,7 @@ static void query_multipage_support() {
         IPC_CREAT | S_IRUSR | S_IWUSR);
       guarantee0(shmid != -1); // Should always work.
       // Try to set pagesize.
-      struct shmid_ds shm_buf = { 0 };
+      struct shmid_ds shm_buf = { {0,0,0,0,0,0,0,0},0,0,0,0,0,0,0,0,0,0,0,0,0,0 };
```

Comment:
Use
```
struct shmid_ds shm_buf{};
```
to _value-initialize_.  Calls the default constructor if there is one. Otherwise, performs _zero-initialization_,
which is what we want here.

Reply:
The final suggested change (to value-initialize the object) seems to have *not* been made.

However, I think it doesn't matter.  The mentioned restriction against being non-empty until C23 is not relevant.
This is C++, not C.  Empty initializers are, and have always been, permitted by C++.

Reply:
Strange the last resulting change I see is
`struct shmid_ds shm_buf{};`

Reply:
That's not what I see in this PR, and not what I see in the repository after the integration.  Where are you seeing
that?  (If in your local repository, maybe you forgot a push to the PR?)

Reply:
I checked it directly in repository
[](https://github.com/openjdk/jdk/blob/master/src/hotspot/os/aix/os_aix.cpp)
And there I see the desired
`struct shmid_ds shm_buf{};`